src/heading.js

```diff
@@ -27,46 +25,120 @@ export default class Heading extends Plugin {
 	 * @inheritDoc
 	 */
 	static get requires() {
-		return [ HeadingEngine ];
+		return [ Paragraph, HeadingEngine ];
 	}
 
 	/**
 	 * @inheritDoc
 	 */
 	init() {
 		const editor = this.editor;
-		const command = editor.commands.get( 'heading' );
-		const options = command.options;
-		const collection = new Collection();
-
-		// Add options to collection.
-		for ( const { id, label } of options ) {
-			collection.add( new Model( {
-				id, label
+		const dropdownItems = new Collection();
+		const options = this._getLocalizedOptions();
+		const commands = [];
+		let defaultOption;
+
+		for ( let option of options ) {
+			// Add the option to the collection.
+			dropdownItems.add( new Model( {
```

Comment:
[Binding requires observable](https://github.com/ckeditor/ckeditor5-ui/blob/master/src/dropdown/list/createlistdropdown.js#L33). We could pass an object here and convert it into observable in createlistdropdown. But this is beyond the scope of this issue as it didn't change this part.